curl/tests/client_test.cc

```diff
@@ -469,15 +473,15 @@ namespace leatherman { namespace curl {
              test_impl->test_failure_mode = curl_impl::error_mode::set_url_error;
              REQUIRE_THROWS_AS(test_client.download_file(req, file_path), curl::http_file_download_exception);
```

Comment:
Hmm, what about:

http_exception:
  -http_request_exception
     -http_setup_exception
     -http_download_exception
        -http_file_exception

My reasoning is that the setup failure results from the passed-in parameter at the end of easy_setopt, which comes from the request. So the request is relevant to that. http_file_exception is part of the download and it may result after a download fails (e.g. when we try to remove the temporary file), so it's important that it preserves the context of the original download exception.